### PR TITLE
scale.py added

### PR DESCRIPTION
**Pull request by shawnhu444**

### CODE REVIEW

File: src/visumorph/scale.py [hunk 1]
```diff
@@ -40,20 +40,16 @@ def scale(image, scale):
...
+    if not isinstance(image, visumorph.Image):
+        raise TypeError("Input 'image' must be a valid VisuMorph Image.")
+    if not isinstance(scale, (int, float)):
+        raise TypeError("Input 'scale' must be a number.")
 
-    if not type(scale) in (float, int):
-        raise TypeError("The scale provided must be a number.")
+    # Calculate new dimensions
+    new_width = round(image.width * scale)
+    new_height = round(image.height * scale)
```
Comment (SoloSynth1) [inline src/visumorph/scale.py]: What happened when the `new_width` and `new_height` is zero or negative? Also, consider the case of `round(0.01)`, will it result in zero?

File: tests/test_rotate.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from visumorph import rotate, load_image
+from visumorph import scale, load_image
```
Comment (SoloSynth1) [inline tests/test_rotate.py]: `test_rotate` should not be related to `scale` so I'm not too sure why is this change here.

File: src/visumorph/scale.py [hunk 3]
```diff
@@ -40,20 +40,16 @@ def scale(image, scale):
...
 
-    if scale <= 0:
-        raise ValueError("The scale provided must be a positive, non-zero "
-                         "number.")
+    # Scale the image
+    scaled_image = image.resize((new_width, new_height))
```
Comment (SoloSynth1) [inline src/visumorph/scale.py]: We do not have `.resize()` method in our `Image` object. Have you ran the tests?

File: tests/test_scale.py
Comment (SoloSynth1) [inline tests/test_scale.py:1]: Please check the pytest results. There are multiple failed tests.

**Review by SoloSynth1 — COMMENTED** [review]
@shawnhu444 Please see comments. Also please note that this PR is duplicate of #46.

**Review by shawnhu444 — COMMENTED** [review]
tests changed made according to comments from Orix

File: tests/test_rotate.py (hunk 2 above)
Comment (shawnhu444) [reply tests/test_rotate.py]: thanks, changes made accordingly

File: src/visumorph/scale.py (hunk 3 above)
Comment (shawnhu444) [reply src/visumorph/scale.py]: thanks, changes made accordingly

File: tests/test_scale.py
Comment (shawnhu444) [reply tests/test_scale.py:1]: thanks, used pytest on the root file and passed local tests

File: tests/test_scale.py [hunk 4]
```diff
@@ -1,40 +1,30 @@
...
     with pytest.raises(TypeError):
-        vm.scale(img, 1.0)
\ No newline at end of file
+        scale(img, 'invalid_scale')
+        
+def test_scale_with_valid_image_should_not_raise_error():
+    img = load_image("tests/img/raw/meme.jpg")
+    try:
+        scale(img, 1.5)
+    except Exception as e:
```
Comment (SoloSynth1) [inline tests/test_scale.py:15]: These tests are not testing for specific error types. Also, as we're using `pytest`, please use `pytest.raises()` which its usages are outlined in the documentation [here](https://docs.pytest.org/en/7.4.x/how-to/assert.html#assertions-about-expected-exceptions) instead of built-in `try`/`except`.

**Comment by shawnhu444** [comment]
scale.py added

File: src/visumorph/scale.py (hunk 1 above)
Comment (SoloSynth1) [reply src/visumorph/scale.py]: Resolving this as this line is removed.

File: tests/test_rotate.py (hunk 2 above)
Comment (SoloSynth1) [reply tests/test_rotate.py]: Resolving this as this change is reverted..

File: src/visumorph/scale.py (hunk 3 above)
Comment (SoloSynth1) [reply src/visumorph/scale.py]: Resolving this as this line is removed.

File: tests/test_scale.py
Comment (SoloSynth1) [reply tests/test_scale.py:1]: Resolving this as tests are passing.

**Review by atabak-alishiri — APPROVED** [review]

**Review by Marcony1 — CHANGES_REQUESTED** [review]
Project too advanced. Deprecated PR.

**Comment by SoloSynth1** [comment]
Closing stale PR.
